diff --git a/src/Composer/Composer.php b/src/Composer/Composer.php
--- a/src/Composer/Composer.php
+++ b/src/Composer/Composer.php
@@ -132,9 +132,17 @@ final class Composer

         $projectAutoload = Reader::read(self::join($this->vendorDir, 'autoload.php'));
         $scipPhpAutoload = Reader::read(self::join($this->scipPhpVendorDir, 'autoload.php'));
-        $autoloadDir = $projectAutoload === $scipPhpAutoload ? $this->scipPhpVendorDir : $this->vendorDir;
-        $loader = require self::join($autoloadDir, 'autoload.php');
-        if (!$loader instanceof ClassLoader) {
+        if ($projectAutoload === $scipPhpAutoload) {
+            $loader = require self::join($this->scipPhpVendorDir, 'autoload.php');
+        } else {
+            $loader = require self::join($this->vendorDir, 'autoload.php');
+            $scipLoader = require self::join($this->scipPhpVendorDir, 'autoload.php');
+            if ($scipLoader instanceof ClassLoader) {
+                $scipLoader->unregister();
+                $scipLoader->register(true);
+            }
+        }
+        if (!($loader instanceof ClassLoader)) {
             throw new RuntimeException("Cannot get autoload.php class loader.");
         }
         $this->loader = $loader;
